src/lib/builders/slider/create.ts

```diff
@@ -338,12 +350,23 @@ export const createSlider = (props?: CreateSliderProps) => {
 				if (val < $min) {
 					updatePosition($min, activeThumbIdx);
 				} else if (val > $max) {
-					updatePosition($max, activeThumbIdx);
+					updatePosition($actualMax, activeThumbIdx);
 				} else {
 					const step = $step;
-					const newValue = mul(Math.round(div(val, step)), step);
-
-					updatePosition(newValue, activeThumbIdx);
+					const min = $min;
+					const actualMax = $actualMax;
+
+					const currentStep = Math.floor((val - min) / step);
+					const midpointOfCurrentStep = min + currentStep * step + step / 2;
+					const midpointOfNextStep = min + (currentStep + 1) * step + step / 2;
+					const newValue =
```

Comment:
I found an extra issue with dragging the slider where it could land on incorrect ticks. I fixed it and ensured that the change happened right between 2 ticks.